Shared/Features/Freaks/View/SkillsView.swift

```diff
@@ -6,16 +6,31 @@ struct SkillsView: View {
 
     var body: some View {
         NavigationView {
-            List {
-                ForEach(viewModel.skills, id: \.self) { skill in
-                    MultipleSelectionRow(viewModel: viewModel, skill: skill, isSelected: viewModel.isSkillSelected(skill: skill)) {
-                        viewModel.setupSelectedSkills(skill: skill)
+            VStack {
+                List {
+                    ForEach(viewModel.skills, id: \.self) { skill in
+                        MultipleSelectionRow(viewModel: viewModel, skill: skill, isSelected: viewModel.isSkillSelected(skill: skill)) {
+                            viewModel.setupSelectedSkills(skill: skill)
+                        }
                     }
                 }
+
+                Button(action: {
+                    self.showSkillsView = false
+                    self.viewModel.isFilterApplied = true
+                }, label: {
+                    Text("Apply")
+                })
+                    .foregroundColor(Color("SecondaryColor"))
+                    .frame(maxWidth: 150, maxHeight: 40)
+                    .background(Color("AccentColor"))
+                    .padding()
+                    .font(.system(size: 18, weight: .bold, design: .default))
             }
             .navigationBarTitle(Text("SKILLS"), displayMode: .inline)
             .navigationBarItems(trailing: Button(action: {
                 viewModel.skillSelections.removeAll()
+                viewModel.isFilterApplied = false
```

Comment:
Now we are resetting this var if we tap on the Reset button, but what happens if we just close the sheet or press the X button?